ddtrace/contrib/aws_lambda/patch.py

```diff
@@ -71,7 +73,8 @@ def _crash_flush(self, _, __):
             root_span.set_tag_str(ERROR_MSG, "Datadog detected an Impending Timeout")
             root_span.set_tag_str(ERROR_TYPE, "Impending Timeout")
         else:
-            log.warning("An impending timeout was reached, but no root span was found. No error will be tagged.")
+            if is_cold_start():
+                log.warning("An impending timeout was reached, but no root span was found. No error will be tagged.")
```

Comment:
non blocking suggestion: To reduce duplication we can can define a LambdaLogger that checks `is_cold_start()` when `log.log(.....)` is called. This would be easier to maintain than making sure every call to `log.warning`, `log.error`, and `log.exception` is nested in a `is_cold_start()` check.  Or at least we can move this check to a helper function:

ex:

```
def lambda_logger(log, level, message):
    if is_cold_start():
        log.(level, message)
```

Reply:
Thanks @mabdinur, I wanted to wait until we had tons of places using this, but since we're already past that (3 times), I will definitely take your sugestion.

Reply:
Will do on another PR after coming back from PTO.